### PR TITLE
Feature/unsaved changes

### PR DESCRIPTION
**Pull request by ezequielc**
@alex-deaconu Please review. Thanks

### CODE REVIEW

File: web/scripts/editor/controllers/ctr-workspace.js [hunk 1]
```diff
@@ -3,13 +3,65 @@
...
+          $timeout(function () {
+            _initializing = false;
+          });
+        } else {
+          $scope.hasUnsavedChanges = true;
+        }
+      }, true);
+
+      $scope.$on('presentationUpdated', function () {
+        $timeout(function () {
```
Comment (alex-deaconu) [inline web/scripts/editor/controllers/ctr-workspace.js]: @ezequielc curious, why do we need a `$timeout` here? Thanks.

File: web/scripts/editor/controllers/ctr-workspace.js [hunk 2]
```diff
@@ -3,13 +3,65 @@
...
       $scope.placeholderFactory = placeholderFactory;
       $scope.isSubcompanySelected = userState.isSubcompanySelected;
       $scope.isTestCompanySelected = userState.isTestCompanySelected;
+      $scope.hasUnsavedChanges = false;
+
+      var _initializing = true;
+      $scope.$watch('factory.presentation', function (newValue, oldValue) {
+        if (_initializing) {
+          $timeout(function () {
+            _initializing = false;
```
Comment (alex-deaconu) [inline web/scripts/editor/controllers/ctr-workspace.js]: @ezequielc why do we need a `$timeout` here as well? Thanks.

**Comment by alex-deaconu** [comment]
@ezequielc one thing I noticed (http://screencast.com/t/y0IsyHFR):

When I add a new Presentation, click Apply on the Presentation settings Modal, and then Cancel, the Save notification doesn't appear. 
However, when I open the Presentation settings and click Apply on an existing Presentation the Save notification does appear. 

Is that because existing Presentations may not have all fields filled in? I don't think it's a big deal, just wanted to bring it up.

Thanks.

**Comment by alex-deaconu** [comment]
@ezequielc same thing happens when I switch between HTML/Layout views.

File: web/scripts/editor/controllers/ctr-workspace.js (hunk 2 above)
Comment (ezequielc) [inline web/scripts/editor/controllers/ctr-workspace.js]: @alex-deaconu `$timeout` code will run just after the `$digest` cycle finishes. When loading an existing presentation, somewhere in the loop the presentation object was being updated, causing the `hasUnsavedChanges` to be flagged.
Using the `$timeout` will make sure `_initializing` is set to false only after the cycle finishes. Thanks

**Comment by ezequielc** [comment]
@alex-deaconu You are right, both cases happen because an existing Presentation does not have all the fields, specially the ones that are purposely not saved as they are used only to help parsing/editor (backgroundStyle, etc). 
I tried to fix it, but I didn't spend much time trying to figure it out as we want to finish Miranda. Should I spend more time and fix it or should we add a card as an improvement for later? Thanks

**Comment by alex-deaconu** [comment]
@ezequielc I don't think either are major issues; just wanted to make sure we're aware of them. We could add a card to prevent the modal from appearing for these particular cases, for our cleanup epic. Maybe a before/after comparison of the Presentation object could help. Either way, not a big deal on either. Thanks.

**Comment by alex-deaconu** [comment]
@ezequielc I'm good with the code in the PR. Also, functionality works as expected, other than the things noted above. Nice work 👍 . Thanks!

**Comment by ezequielc** [comment]
@alex-deaconu I implemented @RV-Adrian 's  suggestion to not notify users when changing template. Please review. Thanks

**Comment by alex-deaconu** [comment]
@ezequielc looks good 👍 . Thanks.

**Pull request merged**